src/app/core/directives/feature-toggle.directive.ts

```diff
@@ -17,34 +19,59 @@ import { FeatureToggleService } from 'ish-core/utils/feature-toggle/feature-togg
 @Directive({
   selector: '[ishFeature]',
 })
-export class FeatureToggleDirective {
+export class FeatureToggleDirective implements OnDestroy {
   private otherTemplateRef: TemplateRef<unknown>;
-  private feature: string;
+  private subscription: Subscription;
+  private enabled$ = new BehaviorSubject<boolean>(undefined);
+  private tick$ = new BehaviorSubject<void>(undefined);
+
+  private destroy$ = new Subject();
 
   constructor(
     private templateRef: TemplateRef<unknown>,
     private viewContainer: ViewContainerRef,
-    private featureToggle: FeatureToggleService
-  ) {}
+    private featureToggle: FeatureToggleService,
+    private cdRef: ChangeDetectorRef
+  ) {
+    combineLatest([
+      this.enabled$.pipe(
+        distinctUntilChanged(),
+        filter(val => typeof val === 'boolean')
```

Comment:
Why do you filter for differently typed values here? I can't see how other values might get in here. Also, you don't do the same check in the `not-feature-toggle`directive.

Reply:
`BehaviorSubject` is initialized with `undefined`